vault/sources/Vault.move

```diff
@@ -236,27 +259,37 @@ module titusvaults::Vault {
             depositor: user_addr,
             amount: coin_value,
             shares_minted: shares_to_mint,
-            round: round_state.current_round_id
+            round: first_round_state.current_round_id
         };
         event::emit(deposit_vault_event);
     }
 
     // instant withdraw vault
-    public (friend) fun instant_withdraw_vault<VaultT, AssetT>(account: &signer, amount: u64) acquires RoundState, Vault, VaultMap {
+    public (friend) fun instant_withdraw_vault<VaultT, AssetT>(account: &signer, amount: u64) acquires Vault, VaultMap {
         let user_addr = address_of(account);
 
-        let round_state = borrow_global_mut<RoundState>(@titusvaults);
         let vault = borrow_global_mut<Vault<VaultT, AssetT>>(@titusvaults);
         let vault_map = borrow_global_mut<VaultMap>(@titusvaults);
-        let user_shares = *smart_table::borrow(&round_state.shares, user_addr);
+
+        // check if there is at least one round in the vault_map.rounds vector
+        assert!(vector::length(&vault_map.rounds) >= 1, E_INVALID_OPERATION);
+
+        let first_round_state = vector::borrow_mut(&mut vault_map.rounds, 0); 
```

Comment:
we need to perform instant withdraw on round ID whose ID has been passed to this function as parameter and not the first round ID.